File: back-end/ms-pet/src/main/java/com/petscare/mspet/model/Pet.java
package com.petscare.mspet.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Data
@Entity
@Table(name="Pets")
public class Pet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private String name;
    @Column
    private String size;
    @Column
    private String race;
    @Column
    private Long clientId;

    @ManyToOne
    @JoinColumn(name = "type_id", referencedColumnName = "id")
    private PetType petType;


    @OneToOne(mappedBy = "pet", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private PetClinicalHistory petClinicalHistory;



}

File: front-end/petscare/src/interfaces/pet.ts
export interface Pet {
    id: number,
    name: string,
    size: string,
    race: string,
    clientId: number
  }
  
  export interface PetFull {
    id: number,
    name: string,
    size: string,
    race: string,
    clientId: number,
    petType: {
      id: number,
      typeName: string
    },
    petClinicalHistory: {
      id: number,
      cratedAt: Date,
      lastUpdate: Date,
      weigth: number
    }
  }